sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSampledSpansProcessor.java

```diff
@@ -0,0 +1,283 @@
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.GuardedBy;
+
+/**
+ * Implementation of the {@link SpanProcessor} that batches spans exported by the SDK the pushes
+ * them to the exporter pipeline.
+ *
+ * <p>All spans reported by the SDK implementation are first added to a synchronized queue (with a
+ * {@code maxQueueSize} maximum size, after the size is reached spans are dropped) and exported
+ * every {@code scheduleDelayMillis} to the exporter pipeline in batches of {@code
+ * maxExportBatchSize}.
+ *
+ * <p>If the queue gets half full a preemptive notification is sent to the worker thread that
+ * exports the spans to wake up and start a new export cycle.
+ *
+ * <p>This batch {@link SpanProcessor} can cause high contention in a very high traffic service.
+ * TODO: Add a link to the SpanProcessor that uses Disruptor as alternative with low contention.
+ */
+public final class BatchSampledSpansProcessor implements SpanProcessor {
+  private static final String WORKER_THREAD_NAME = "BatchSampledSpansProcessor_WorkerThread";
+  private final Worker worker;
+  private final Thread workerThread;
+
+  private BatchSampledSpansProcessor(
+      SpanExporter spanExporter,
+      long scheduleDelayMillis,
+      int maxQueueSize,
+      int maxExportBatchSize) {
+    this.worker = new Worker(spanExporter, scheduleDelayMillis, maxQueueSize, maxExportBatchSize);
+    this.workerThread = newThread(worker);
+    this.workerThread.start();
+  }
+
+  @Override
+  public void onStartSync(ReadableSpan span) {}
+
+  @Override
+  public void onEndSync(ReadableSpan span) {
+    if (!span.getSpanContext().getTraceOptions().isSampled()) {
+      return;
+    }
+    worker.addSpan(span);
+  }
+
+  @Override
+  public void shutdown() {
+    workerThread.interrupt();
+    worker.flush();
+  }
+
+  /**
+   * Returns a new Builder for {@link BatchSampledSpansProcessor}.
+   *
+   * @param spanExporter the {@code SpanExporter} to where the sampled Spans are pushed.
+   * @return a new {@link BatchSampledSpansProcessor}.
+   * @throws NullPointerException if the {@code spanExporter} is {@code null}.
+   */
+  public static Builder newBuilder(SpanExporter spanExporter) {
+    return new Builder(spanExporter);
+  }
+
+  /** Builder class for {@link BatchSampledSpansProcessor}. */
+  public static final class Builder {
+    private final SpanExporter spanExporter;
+    private long scheduleDelayMillis = 5000;
+    private int maxQueueSize = 2048;
+    private int maxExportBatchSize = 512;
+
+    private Builder(SpanExporter spanExporter) {
+      this.spanExporter = Utils.checkNotNull(spanExporter, "spanExporter");
+    }
+
+    // TODO: Consider to add support for constant Attributes and/or Resource.
+
+    /**
+     * Sets the delay interval between two consecutive exports. The actual interval may be shorter
+     * if the batch size is getting larger than {@code maxQueuedSpans / 2}.
```

Comment:
You can also mention the default values. Same for other setters.

Reply:
Done.